create_infotexts.py

```diff
@@ -239,12 +216,57 @@ def __init__(self):
         self.content_cats = []  # content cateogories without 'Category:'-prefix
         self.meta_cats = []  # maintance categories without 'Category:'-prefix
         self.data = {}  # dictionary holding individual field values as wikitext
+        self.filename = ""  # without filename extension
 
         batch_cats = ["Swedish Cyprus Expedition",
                       "Media_contributed_by_SMVK_2017-02"]
 
         self.meta_cats.extend(batch_cats)
 
+    def create_commons_filename(self, item):
+        """
+        Transform original filename into Wikimedia commons style filename.
+
+        See https://phabricator.wikimedia.org/T156612 for definition.
+        Note: the returned filename does not include file extension e.g. '.tif'
+
+        :param item: dictionary conatining metadata for one image
+        :return: None (populates self.filename)
+        """
+        fname_str = ""
+        fname_str += re.sub(" Svenska Cypernexpeditionen\.?", "", item["Beskrivning"])
+        if not fname_str.endswith(" "):
+            fname_str += " "
+
+        # Enrich with keywords
+        if self.data["keyword_list"]:
+            for keyword in self.data["keyword_list"]:
+                if keyword not in item["Beskrivning"]:
+                    fname_str += keyword + " "
```

Comment:
Do we not want them comma separated?
Would it be easier to construct a "keywords to add list" and then add them all in the end

```python
# Enrich with keywords
keywords_to_add = []
if self.data["keyword_list"]:
    for keyword in self.data["keyword_list"]:
        if keyword not in item["Beskrivning"]:
            keywords_to_add.append(keyword)
```
that way you also have more control over when you add the text to `fname_str`